src/gui/inputdevices/TouchInputHandler.cpp

```diff
@@ -137,16 +132,25 @@ void TouchInputHandler::zoomStart() {
         zoomControl->setZoomFitMode(false);
     }
 
+    auto center = (this->priLastAbs + this->secLastAbs) / 2.0;
+
     auto* mainWindow = inputContext->getView()->getControl()->getWindow();
+    // use screen pixel coordinates for the zoom center
+    // as relative coordinates depend on the changing zoom level
+    int rx, ry;
+    gdk_window_get_root_coords(gtk_widget_get_window(mainWindow->getWinXournal()), 0, 0, &rx, &ry);
+    center.x -= rx;
+    center.y -= ry;
 
     // When not using touch drawing, we're using a different scrolling method.
     // This requires different centering.
-    if (!mainWindow->getGtkTouchscreenScrollingEnabled()) {
-        Rectangle zoomSequenceRectangle = zoomControl->getVisibleRect();
-        center += utl::Point<double>{zoomSequenceRectangle.x, zoomSequenceRectangle.y};
-    }
+    /*if (!mainWindow->getGtkTouchscreenScrollingEnabled()) {
+        center = (this->priLastRel + this->secLastRel) / 2.0;  // TODO check!
+    }*/
```

Comment:
```suggestion
```

Reply:
Please also remove the comment above these lines.